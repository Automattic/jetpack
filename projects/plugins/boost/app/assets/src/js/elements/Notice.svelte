<script lang="ts">
	import { Button, Notice } from '@automattic/jetpack-components';
	import React from 'react';
	import { slide } from 'svelte/transition';
	import { dismissedPopOuts } from '../stores/config';
	import ReactComponent from './ReactComponent.svelte';

	type ActionButton = {
		label: string;
		onClick?: () => void;
		isLoading?: boolean;
		disabled?: boolean;
	};

	export let level: 'info' | 'warning' | 'success' | 'error' = 'info';
	export let title: string;
	export let message: string;
	export let actions: ActionButton[] | undefined;
	export let dismissalKey: string | undefined;

	$: isDismissed = dismissalKey && $dismissedPopOuts.includes( dismissalKey );

	$: actionComponents =
		! isDismissed &&
		actions?.map( ( action, index ) =>
			React.createElement(
				Button,
				{
					isPrimary: true,
					onClick: action.onClick,
					key: index,
					isLoading: !! action.isLoading,
					disabled: !! action.disabled,
				},
				action.label
			)
		);

	function dismiss() {
		dismissedPopOuts.dismiss( dismissalKey );
	}
</script>

{#if ! isDismissed}
	<div transition:slide|local>
		<ReactComponent
			this={Notice}
			{level}
			{title}
			children={message}
			actions={actionComponents}
			onClose={dismissalKey ? dismiss : undefined}
			hideCloseButton={! dismissalKey}
		/>
	</div>
{/if}
