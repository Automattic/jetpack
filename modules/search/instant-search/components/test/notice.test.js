/** @jsx h
 * @jest-environment jsdom
 */
/* global expect */

/**
 * External dependencies
 */
import { h } from 'preact';
import { render } from '@testing-library/preact';
import '@testing-library/jest-dom/extend-expect';

/**
 * Internal dependencies
 */
import Notice from '../notice';

test( 'returns a notice if the type is warning', () => {
	const { asFragment } = render( <Notice type="warning" /> );
	expect( asFragment() ).toMatchSnapshot();
} );
