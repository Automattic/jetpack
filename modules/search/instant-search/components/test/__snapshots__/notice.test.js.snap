// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a notice if the type is warning 1`] = `
<DocumentFragment>
  <div
    class="jetpack-instant-search__notice jetpack-instant-search__notice--warning"
  >
    <svg
      class="gridicon gridicons-info "
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
        />
      </g>
    </svg>
  </div>
</DocumentFragment>
`;
